### PR TITLE
gateways: add Gateway and Route config entry status condition types and helpers

### PR DESCRIPTION
**Pull request by mikemorris**
### Links
- https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.23/#condition-v1-meta
- https://github.com/kubernetes-sigs/gateway-api/blob/main/apis/v1beta1/shared_types.go

### PR Checklist

* [ ] updated test coverage
* [ ] external facing docs updated
* [x] not a security concern

### CODE REVIEW

File: agent/consul/fsm_data_store.go [hunk 1]
```diff
@@ -0,0 +1,77 @@
+package consul
```
Comment (sarahalsmiller) [inline agent/consul/fsm_data_store.go:1]: You'll probably want to rebase here? I don't think you meant to touch these files.

File: proto/pbconfigentry/config_entry.proto [hunk 2]
```diff
@@ -602,6 +602,7 @@ message Status {
 // name=Structs
 message Condition {
   string Type = 1;
+  // mog: func-from=conditionStatusFromStructs func-to=conditionStatusToStructs
```
Comment (mikemorris) [inline proto/pbconfigentry/config_entry.proto:605]: Why does this not seem to be working, do I need to regenerate something?

Comment (mikemorris) [reply proto/pbconfigentry/config_entry.proto:605]: Needed to reinstall Xcode developer tools to get `make tools` and then `make proto` to run successfully.

**Comment by mikemorris** [comment]
Superceded by https://github.com/hashicorp/consul/pull/16994

**Pull request closed without merging**